src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -187,7 +191,41 @@ private IEnumerable<MongoUpdate> ConvertUpdateEntriesToMongoUpdates(IList<IUpdat
 
     private static MongoUpdate ConvertAddedEntryToMongoUpdate(string collectionName, string id, IUpdateEntry entry)
     {
-        throw new NotImplementedException(); // EF-17
+        var entityType = entry.EntityType;
+        var entitySerializer = (IBsonDocumentSerializer)EntitySerializer.Create(entityType);
+
+        var document = new BsonDocument();
+        using (var writer = new BsonDocumentWriter(document))
+        {
+            writer.WriteStartDocument();
+
+            foreach (var property in entityType.GetProperties())
+            {
+                var propertyValue = entry.GetCurrentValue(property);
+                var propertySerializationInfo = GetPropertySerializationInfo(entitySerializer, property);
+                var elementName = propertySerializationInfo.ElementName;
+                var propertySerializer = propertySerializationInfo.Serializer;                   
+                var context = BsonSerializationContext.CreateRoot(writer);
+
+                writer.WriteName(elementName);
+                propertySerializer.Serialize(context, propertyValue);
```

Comment:
You talked about converters... but I don't know what those are, and in any case in the MongoDB C# driver world it is the serializers that determine how things are stored in the database.